cachito/web/models.py

```diff
@@ -893,35 +866,14 @@ def _state_to_json(state):
             latest_state = states[0]
             rv["state_history"] = states
 
-            package_to_deps = {}
-            # This is used to quickly see if a dependency has been added to the root "dependencies"
-            # array in the returned JSON. This uses a set instead of iterating through the
-            # "dependencies" array to avoid an O(n) operation for every dependency.
-            seen_dependencies = set()
-            rv["dependencies"] = []
-            for req_dep in self.request_dependencies:
-                dep_json = req_dep.dependency.to_json(
-                    req_dep.replaced_dependency.to_json() if req_dep.replaced_dependency else None,
-                    force_replaces=True,
-                )
-                # Avoid duplicate dependencies in the root "dependencies" array
-                if req_dep.dependency.id not in seen_dependencies:
-                    rv["dependencies"].append(dep_json)
-                    seen_dependencies.add(req_dep.dependency.id)
-
-                package_to_deps.setdefault(req_dep.package.id, []).append(dep_json)
-
-            sort_packages_and_deps_in_place(rv["dependencies"])
-
-            rv["packages"] = [
-                req_package.package.to_json(
-                    dependencies=package_to_deps.get(req_package.package_id, []),
-                    subpath=req_package.subpath,
-                )
-                for req_package in self.request_packages
-            ]
-
-            sort_packages_and_deps_in_place(rv["packages"])
+            bundle_dir = RequestBundleDir(
+                self.id, root=flask.current_app.config["CACHITO_BUNDLES_DIR"]
+            )
+            packages_data = PackagesData()
+            packages_data.load(bundle_dir.packages_data)
+            packages_data.sort()
```

Comment:
Technically we don't need to sort here anymore, when loading a single sorted file, the packages data will be sorted.

We included the sorting changes in the same release as the packages.json generation, so there should not be any unsorted files in any of the Cachito environments.